gemini/sample-apps/gemini-mesop-cloudrun/main.py

```diff
@@ -0,0 +1,1650 @@
+    s.radio_value = event.value
+
+
+def on_length_radio_change(event: me.RadioChangeEvent) -> None:
+    s = me.state(State)
+    s.story_length_value = event.value
+
+
+def generate_story(e: me.ClickEvent | me.EnterEvent) -> None:
+    s = me.state(State)
+    s.story_output = "" # clear any existing story
+    s.story_progress = True
+    yield
+
+    if s.story_temp_value == "low":
+        temp = 0.5
+    elif s.story_temp_value == "high":
+        temp = 1.5
+    else:
+        temp = 1
+    prompt = f"""Write a {s.story_length_value} story based on the following premise: \n
+    character_name: {s.story_character_name} \n
+    character_type: {s.story_character_type} \n
+    character_persona: {s.story_character_personality} \n
+    character_location: {s.story_character_location} \n
+    story_premise: {",".join(s.story_selected_premises)} \n
+    If the story is "short", then make sure to have 5 chapters or else if it is "long" then 10 chapters.
+    Important point is that each chapters should be generated based on the premise given above.
+    First start by giving the book introduction, chapter introductions and then each chapter. It should also have a proper ending.
+    The book should have prologue and epilogue.
+    """
+    print(f"prompt: {prompt}")
+
+    model = GenerativeModel(s.model)
+    config = GenerationConfig(temperature=temp, max_output_tokens=2048)
+    safety_settings = {
+        HarmCategory.HARM_CATEGORY_HARASSMENT: HarmBlockThreshold.BLOCK_NONE,
+        HarmCategory.HARM_CATEGORY_HATE_SPEECH: HarmBlockThreshold.BLOCK_NONE,
+        HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: HarmBlockThreshold.BLOCK_NONE,
+        HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_NONE,
+    }
+    response = model.generate_content(
+        prompt,
+        generation_config=config,
+        safety_settings=safety_settings,
+        )
+    print(response)
+    s.story_output = response.text
+    s.story_progress = False
+    yield
+
+
+## Marketing events
+def on_change_marketing_category(e: me.RadioChangeEvent) -> None:
+    state = me.state(State)
+    state.marketing_product_category = e.value
+
+
+def on_change_marketing_target(e: me.RadioChangeEvent) -> None:
+    state = me.state(State)
+    state.marketing_target_audience = e.value
+
+
+def on_change_marketing_location(e: me.RadioChangeEvent) -> None:
+    state = me.state(State)
+    state.marketing_target_location = e.value
+
+
+def on_selection_change_marketing_goals(e: me.SelectSelectionChangeEvent) -> None:
+    s = me.state(State)
+    s.marketing_campaign_selected_goals = e.values
+    print(f"selected: {s.marketing_campaign_selected_goals}")
+
+
+def on_change_marketing_brand_voice(e: me.RadioChangeEvent) -> None:
+    state = me.state(State)
+    state.marketing_brand_voice = e.value
+
+
+def on_change_marketing_budget(e: me.RadioChangeEvent) -> None:
+    state = me.state(State)
+    state.marketing_budget = e.value
```

Comment:
Is there any way to simplify this into fewer methods?

Reply:
Can do - will combine single radio settings to a setattr(state, e.key, e.value)